File: ride-service/src/main/java/com/internship/rideservice/service/communication/ArtemisConsumer.java
package com.internship.rideservice.service.communication;

import com.internship.commonevents.event.ChangeRideStatusEvent;
import com.internship.rideservice.service.command.CommandRideService;
import lombok.RequiredArgsConstructor;
import org.springframework.jms.annotation.JmsListener;
import org.springframework.stereotype.Service;

import java.util.logging.Logger;

@Service
@RequiredArgsConstructor
public class ArtemisConsumer {
    private final CommandRideService commandRideService;
    Logger logger = Logger.getLogger(ArtemisConsumer.class.getName());

    @JmsListener(destination = "ride-status-updates")
    public void receiveRideStatusUpdate(ChangeRideStatusEvent event) {
        try {
            logger.info("Received change ride status event: " + event.toString());
            commandRideService.changeRideStatus(event);
        } catch (Exception e) {
            throw new RuntimeException("Failed to deserialize message", e);
        }
    }
}

File: finance-service/src/main/kotlin/com/internship/financeservice/service/ArtemisConsumer.kt
package com.internship.financeservice.service

import com.internship.commonevents.event.ConfirmedPaymentRequest
import com.internship.financeservice.service.finance.CommandFinanceOperationService
import org.springframework.context.annotation.Configuration
import org.springframework.jms.annotation.JmsListener

@Configuration
class ArtemisConsumer (
    private val commandFinanceOperationService: CommandFinanceOperationService
){

    @JmsListener(destination = "payment-confirmation")
    fun receiveCashPaymentConfirmation(event: ConfirmedPaymentRequest) {
        try {
            println("recieved " + event)
            commandFinanceOperationService.createPaymentByCash(event)
        } catch (e: Exception) {
            throw RuntimeException("Failed to deserialize message", e)
        }
    }

}